src/topology/homotopy/homotopy_group.lean

```diff
@@ -5,141 +5,301 @@ Authors: Roberto Alvarez
 -/
 
 import algebraic_topology.fundamental_groupoid.fundamental_group
+import group_theory.eckmann_hilton
+import logic.equiv.transfer_instance
+import algebra.group.ext
 
 /-!
 # `n`th homotopy group
 
-We define the `n`th homotopy group at `x`, `π n x`, as the equivalence classes
+We define the `n`th homotopy group at `x`, `π_n x`, as the equivalence classes
 of functions from the nth dimensional cube to the topological space `X`
 that send the boundary to the base point `x`, up to homotopic equivalence.
 Note that such functions are generalized loops `gen_loop n x`, in particular
 `gen_loop 1 x ≃ path x x`
 
-We show that `π 0 x` is equivalent to the path-conected components, and
-that `π 1 x` is equivalent to the fundamental group at `x`.
+We show that `π_0 x` is equivalent to the path-conected components, and
+that `π_1 x` is equivalent to the fundamental group at `x`.
+We give a group instance using path composition and show commutativity when `n > 1`.
 
 ## definitions
 
-* `gen_loop n x` is the type of continous fuctions `I^n → X` that send the boundary to `x`
-* `homotopy_group n x` denoted `π n x` is the quotient of `gen_loop n x` by homotopy relative
-  to the boundary
+* `gen_loop n x` is the type of continous fuctions `I^n → X` that send the boundary to `x`,
+* `homotopy_group n x` denoted `π_ n x` is the quotient of `gen_loop n x` by homotopy relative
+  to the boundary,
+* group instance `group (π_(n+1) x)`,
+* commutative group instance `comm_group (π_(n+2) x)`.
 
-TODO: show that `π n x` is a group when `n > 0` and abelian when `n > 1`. Show that
-`pi1_equiv_fundamental_group` is an isomorphism of groups.
+TODO:
```

Comment:
`Ω^n (Ω^m X) \equiv Ω^(n+m) X` is also a good target

Reply:
Should probably use `sum N M`. And separately `Ω^N X \equiv Ω^M X` given `N \equiv M`.